pkg/reconciler/ingress/lister.go

```diff
@@ -83,13 +82,13 @@ func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1al
 			qualifiedTarget := status.ProbeTarget{
 				PodIPs: target.PodIPs,
 				Port:   target.Port,
-				URLs:   []neturl.URL{},
+				URLs:   make([]*url.URL, len(gatewayHosts[gatewayName])),
 			}
-			// Use sorted host for consistent ordering.
-			for _, host := range gatewayHosts[gatewayName].List() {
-				newURL := target.URLs[0]
+			// Use sorted hosts list for consistent ordering.
+			for i, host := range gatewayHosts[gatewayName].List() {
+				newURL := *target.URLs[0]
```

Comment:
Can we skip dereferencing a pointer here and using `&` line 91?

Reply:
No, since we need a clone to keep source immutable.

Reply:
:+1: